src/Illuminate/Collections/Arr.php

```diff
@@ -731,13 +731,13 @@ public static function random($array, $number = null, $preserveKeys = false)
             );
         }
 
-        if (empty($array) || (! is_null($number) && $number <= 0)) {
+        if (empty($array) || ($number !== null && $number <= 0)) {
             return is_null($number) ? null : [];
```

Comment:
You might have missed a case here:
```suggestion
            return $number === null ? null : [];
```